File: 3788-MaximumUniqueSubarraySumAfterDeletion/3788-MaximumUniqueSubarraySumAfterDeletion.java
// Last updated: 7/30/2025, 11:57:00 AM
class Solution {
    public int maxSum(int[] nums) {
        Arrays.sort(nums);
        if(nums[nums.length -1] < 0){
            return nums[nums.length -1];
        }
        Set<Integer> n = new HashSet<>();
        int sum = 0;
        for(int i=0; i<nums.length; i++){
            if(!(n.contains(nums[i])) && nums[i]>0){
                n.add(nums[i]);
                sum += nums[i];
            }
        }
        return sum;
    }
}

File: 3788-MaximumUniqueSubarraySumAfterDeletion/3788-MaximumUniqueSubarraySumAfterDeletion.py
# Last updated: 7/30/2025, 11:57:07 AM
class Solution:
    def maxSum(self, nums: List[int]) -> int:
        if(max(nums)<0):
            return max(nums)
        n = []
        sums = 0
        for i in nums:
            if i not in n and i>0:
                n.append(i)
                sums += i
        return sums
        